posts/2024-08-07-Project-goals.md

```diff
@@ -0,0 +1,89 @@
+
+[RFC #3672]: https://github.com/rust-lang/rfcs/pull/3672#issuecomment-2254599176
+[RFC #3614]: https://github.com/rust-lang/rfcs/pull/3614
+
+## Rust’s mission
+
+Ultimately, all the goals selected are intended to further Rust's mission of **empowering everyone to build reliable and efficient software**. Rust targets programs that prioritize
+
+* reliability and robustness;
+* performance, memory usage, and resource consumption; and
+* long-term maintenance and extensibility.
+
+We consider "any two out of the three" as the right heuristic for projects where Rust is a strong contender or possibly the best option.
+
+## Why these particular flagship goals?
+
+**2024 Edition.** 2024 will mark the 4th Rust edition, following on the 2015, 2018, and 2021 editions. Similar to the [2021 edition](https://rust-lang.github.io/rust-project-goals/2024h2/https://github.com/nikomatsakis/rfcs/blob/rfl-project-goal/text/3085-edition-2021.html), the 2024 edition is not a "major marketing push" but rather an opportunity to correct small ergonomic issues with Rust that will make it overall much easier to use. The changes planned for the 2024 edition will (1) support `-> impl Trait` and `async fn` in traits by aligning capture behavior; (2) permit (async) generators to be added in the future by reserving the `gen` keyword; and (3) alter fallback for the `!` type. The [plan] is to finalize development of 2024 features this year; the Edition itself is planned for Rust v1.85 (released to beta 2025-01-03 and to stable on 2025-02-20).
+
+**Async.** In 2024 we plan to deliver several critical async Rust building block features, most notably support for *async closures* and *`Send` bounds*. This is part of a multi-year program aiming to raise the experience of authoring "async Rust" to the same level of quality as "sync Rust". Async Rust is a crucial growth area, with 52% of the respondents in the [2023 Rust survey](https://blog.rust-lang.org/2024/02/19/2023-Rust-Annual-Survey-2023-results.html) indicating that they use Rust to build server-side or backend applications. 
+
+**Rust for Linux.** The [experimental support for Rust development in the Linux kernel][RFL.com] is a watershed moment for Rust, demonstrating to the world that Rust is indeed capable of targeting all manner of low-level systems applications. And yet today that support rests on a [number of unstable features][RFL#2], blocking the effort from ever going beyond experimental status. For 2024H2 we will work to close the largest gaps that block support, such as 
+
+[RFL.com]: https://rust-for-linux.com/
```

Comment:
I don't think we should hide the real URL like this, it opens the door to confusion and/or attacks by someone squatting the short form.

Reply:
This is just a Markdown alias, right? `RFL.com` won't ever appear as a link or text in the blog post, it's just an internal shortcut to reference the correct link. The link text will be "experimental support for Rust development in the Linux kernel".

Reply:
Perhaps Mark meant in the source view? i.e. someone may open the file and try to "click-to-go" it (I definitely had to look closely for a moment since it caught my attention). Removing the `.com` part would avoid that kind of issue.

Reply:
OK, changed